addons/web/static/src/legacy/js/widgets/colorpicker.js

```diff
@@ -44,7 +44,16 @@ var ColorpickerWidget = Widget.extend({
         // anywhere on the screen, crossing iframes).
         // TODO adapt in master: these events should probably be bound in
         // `start` instead of `init` (at least to be more conventional).
-        this.$documents = $([window.top, ...Array.from(window.top.frames)].map(w => w.document));
+        this.$documents = $([window.top, ...Array.from(window.top.frames)].filter(frame => {
+            try {
+                const document = frame.document;
+                return !!document;
+            } catch (error) {
+                // We cannot access the document (cross origin) or there is no
+                // document.
+                return false;
+            }
+        }).map(w => w.document));
```

Comment:
@Guillaume-gdi Thanks for the quick reaction :+1: 

> The fix isn't very elegant but I can't think of a better way of doing it... maybe you'll have a better idea?

I think it is nice :+1: Just: your filter checks `window.top` too, which is unnecessary. `window.top` can only be the Odoo instance page as Odoo backends cannot be embedded inside an iframe (from another origin). You cannot even be connected to an Odoo website from inside an iframe. However, I guess it is a possibility that someone uses that colorpicker widget in a website page custom code, to be used by unconnected website visitors and that this website page could be embedded into an iframe... but I think I am willing to take the bet this is not something which currently exists this once :laughing: So could just apply the filter on `Array.from(window.top.frames)` if you agree? :+1: